WalletWasabi/Affiliation/CoinJoinRequestsUpdater.cs

```diff
@@ -0,0 +1,247 @@
+using System.Linq;
+using System.Collections.Immutable;
+using Microsoft.Extensions.Hosting;
+using WalletWasabi.Affiliation.Models;
+using NBitcoin;
+using WalletWasabi.WabiSabi.Backend.Rounds;
+using System.Threading;
+using System.Threading.Tasks;
+using WalletWasabi.WabiSabi.Backend.Events;
+using System.Collections.Concurrent;
+using WalletWasabi.Affiliation.Extensions;
+using WalletWasabi.Affiliation.Models.CoinjoinRequest;
+using WalletWasabi.Logging;
+
+namespace WalletWasabi.Affiliation;
+
+public class CoinJoinRequestsUpdater : BackgroundService
+{
+	private static readonly TimeSpan AffiliateServerTimeout = TimeSpan.FromSeconds(60);
```

Comment:
I see, it looks cleaner but also a bit riskier if we try to break that dependency. Anyway, it would be good to consider doing that and assign it a priority (low for now)

Reply:
Now the `CoinJoinRequestsUpdater` doesn't really depends on arena anymore but on a class that exposes this method only:

`
IAsyncEnumerable<RoundNotification> GetFinalizedRounds(CancellationToken cancellationToken)
`

So, i think that can make it easier to re-wire it.